vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java

```diff
@@ -509,6 +512,16 @@ public void handle(Buffer event) {
         BiConsumer<InputChannel, Object> readCallback = null;
         Object context = null;
         synchronized (request.connection()) {
+            uploadSize += event.length();
+            if (maxEntitySize != UndertowOptions.DEFAULT_MAX_ENTITY_SIZE && uploadSize > maxEntitySize) {
+                responseDone = true;
+                terminateRequest();
+                response.setStatusCode(413);
```

Comment:
This should probably check if a response has already been sent or not, and if so just call .close(). It also should set the Connection: close header.

Reply:
Thanks for the review @stuartwdouglas. I have addressed this part.